bagua/torch_api/contrib/cache_loader.py

```diff
@@ -0,0 +1,135 @@
+import pickle
+from collections import defaultdict
+
+__all__ = ["CacheLoader"]
+
+
+def serialize(input):
+    return pickle.dumps(input)
+
+
+def deserialize(input):
+    return pickle.loads(input)
+
+
+class CacheLoader:
+    def __init__(
+        self,
+        backend: str = "redis",
+        key_prefix: str = "",
+        writer_buffer_size: int = 1,
+        **kwargs,
+    ):
+        """
+        A mapping from keys to values. Values are automatically loaded by the cache, and
```

Comment:
`CacheLoader` caches values calculated by an expensive function by theirs keys via `CacheLoader.get` method, so that the values can be retrieved faster next time.